### PR TITLE
optimizing the front-end hash

### PR DESCRIPTION
**Pull request by happygaijin**

### CODE REVIEW

File: src/lib/helpers/clientInfo.ts [hunk 1]
```diff
@@ -19,32 +19,43 @@ export interface ClientAgentInfo {
     canvasPrint?: string
 }
 
+const toSha256Hash = async (data): Promise<string> => {
+    const encoder = new TextEncoder();
+    const hash = await crypto.subtle.digest('SHA-256', encoder.encode(data));
```
Comment (cjkoepke) [inline src/lib/helpers/clientInfo.ts:24]: @happygaijin `crypto` library needs to be imported.

**Review by cjkoepke — CHANGES_REQUESTED** [review]

File: src/lib/helpers/clientInfo.ts (hunk 1 above)
Comment (happygaijin) [reply src/lib/helpers/clientInfo.ts:24]: I am pretty sure this is a native api. I just opened an empty browser window console and this code works. But you can double-check that for me. I didn't have to import anything for this to work locally if that helps.  
https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto

Comment (cjkoepke) [reply src/lib/helpers/clientInfo.ts:24]: Holy cow, can't believe I didn't know about this. Pretty incredible. It's not supported in IE, but I'm gonna say we don't care about that (its unsupported anymore anyway).

**Review by cjkoepke — APPROVED** [review]
LGTM

**Pull request merged**